bot/exts/halloween/spooknamerate.py

```diff
@@ -0,0 +1,329 @@
+                winners.append(winner)
+                if len(winner_messages) > i + 1:
+                    if winner_messages[i + 1][1]['score'] != winner[1]['score']:
+                        break
+                elif len(winner_messages) == (i + 1) + 1:  # The next element is the last element and len()
+                    # returns the entire length
+                    if winner_messages[i + 1][1]['score'] != winner[1]['score']:
+                        break
+
+            # one iteration is complete
+            await channel.send("Today's Spook Name Rate Game ends now, and the winner(s) is(are)...")
+
+            async with channel.typing():
+                await asyncio.sleep(1)  # give the drum roll feel
+
+                if len(winners) > 0:  # make sure there is a winner
+
+                    if len(winners) > 1:  # if there are more than one winners
+                        await channel.send(" and ".join([win[1]['author'].mention for win in winners]) + " !")
+                        score = winners[0][1]['score']
+                        # display winners, their words and scores
+                        await channel.send(f"Congratulations to all! You have a score of {score}!")
+                        wrds = ', '.join([f'{win[1]["word"]} ({win[1]["author"].mention})' for win in winners])
+                        await channel.send(f"Your words were: {wrds}!")
+
+                    else:  # if there is only one winner
+                        # There cant be 0 winners since we are checking that above
+
+                        winner = winners[0][1]
+                        await channel.send(f"{winner['author'].mention}!")
+                        await channel.send(f"Congratulations to {winner['author'].mention}! You have a score of \
+{winner['score']}!")
+                        await channel.send(f"Your word was: **{winner['word']}**.")  # display winner, word and score
+
+                    # Send random party emojis
+                    party = [random.choice([':partying_face:', ':tada:']) for _ in range(random.randint(1, 10))]
+                    await channel.send(" ".join(party))
+
+                else:  # if there isn't a winner
+                    # no participants in current iteration
+                    await channel.send('Hmm... Looks like no one participated! :cry:')
+
+            self.messages = {}  # reset the messages
+
+        # send the next name
+        self.word = random.choice(self.first_names) + " " + random.choice(self.last_names)
+        async with channel.typing():
+            for message in ["Anyways... let's move on to the next name!",   # starts new game
+                            "And the next name is...",  # And today's name is...
+                            f"**{self.word}**!",
+                            "Try to spookify that... :smirk:"]:
+                await asyncio.sleep(0.7)
+                await channel.send(message)
```

Comment:
Don't send messages in loop. It might look cool, but it's borderline api abuse, even with the sleep.
Just format one message and send that, even if it doesn't look as cool.

Reply:
okay, I have fixed it.